lib/Parse/ParseExpr.cpp

```diff
@@ -836,9 +836,11 @@ static StringRef copyAndStripUnderscores(ASTContext &C, StringRef orig) {
   char *start = static_cast<char*>(C.Allocate(orig.size(), 1));
   char *p = start;
```

Comment:
It doesn't look like `p` could ever be null here, unless allocation failed. `*p++` does not present a risk of `bool(p)` becoming false.